src/FileUpload/FileUpload.js

```diff
@@ -21,14 +21,7 @@ class FileUpload extends React.PureComponent {
 
     return (
       <label className={className} data-hook={dataHook}>
-        {React.cloneElement(children, {
-          onClick: event => {
-            if (typeof children.props.onClick === 'function')
-              children.props.onClick();
-            this.inputRef.current.click();
-            event.preventDefault();
-          },
-        })}
+        {children({ upload: () => this.inputRef.current.click() })}
```

Comment:
`upload` does not indicate what this function really does.
I suggest `openFileUploadDialog`:

```suggestion
        {children({ openFileUploadDialog: () => this.inputRef.current.click() })}
```